Sources/MongoSwift/MongoError.swift

```diff
@@ -2,277 +2,287 @@ import CLibMongoC
 import Foundation
 
 /// An empty protocol for encapsulating all errors that this package can throw.
-public protocol MongoError: LocalizedError {}
-
-// TODO: update this link and the one below (SWIFT-319)
-/// A MongoDB server error code.
-/// - SeeAlso: https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml
-public typealias ServerErrorCode = Int
+public protocol MongoErrorProtocol: LocalizedError {}
```

Comment:
to disambiguate from the namespace, I added the `Protocol` suffix that I've seen in a few places in the stdlib (e.g. [`LazySequenceProtocol`](https://developer.apple.com/documentation/swift/lazysequenceprotocol))